operator/pkg/controllers/master.go

```diff
@@ -75,3 +83,61 @@ func (m *MasterManager) createPod(r *ElasticJobReconciler, job *elasticv1alpha1.
 	}
 	return nil
 }
+
+func (m *MasterManager) createPodName(job *elasticv1alpha1.ElasticJob) string {
+	return fmt.Sprintf("%s-%s", job.GetName(), string(ReplicaTypeEasydlMaster))
+}
+
+func (m *MasterManager) syncMasterState(r *ElasticJobReconciler, job *elasticv1alpha1.ElasticJob) error {
+	master, err := m.getMasterPod(r, job)
+	if err != nil {
+		logger.Warnf("Failed to get master, error : %v", err)
+	}
+	if master == nil {
+		return nil
+	}
+	if master.Status.Phase == corev1.PodSucceeded {
+		job.Status.ReplicaStatuses[ReplicaTypeEasydlMaster].Succeeded = 1
+		msg := fmt.Sprintf("job(%s/%s) successfully completed", job.Namespace, job.Name)
+		r.Recorder.Event(job, corev1.EventTypeNormal, string(commonv1.JobSucceeded), msg)
+		if job.Status.CompletionTime == nil {
+			now := metav1.Now()
+			job.Status.CompletionTime = &now
+		}
+		updateStatus(&job.Status, commonv1.JobSucceeded, commonutil.JobCreatedReason, msg)
+	} else if master.Status.Phase == corev1.PodFailed {
+		job.Status.ReplicaStatuses[ReplicaTypeEasydlMaster].Failed = 1
+		msg := fmt.Sprintf("job(%s/%s) has failed", job.Namespace, job.Name)
+		reason := master.Status.Reason
+		if reason == "" {
+			reason = commonutil.JobFailedReason
+		}
+		r.Recorder.Event(job, corev1.EventTypeWarning, reason, msg)
+		if job.Status.CompletionTime == nil {
+			now := metav1.Now()
+			job.Status.CompletionTime = &now
+		}
+		updateStatus(&job.Status, commonv1.JobFailed, reason, msg)
+	} else if master.Status.Phase == corev1.PodRunning || master.Status.Phase == corev1.PodPending {
+		job.Status.ReplicaStatuses[ReplicaTypeEasydlMaster].Active = 1
+		if !isRunning(job.Status) {
+			msg := fmt.Sprintf("job(%s/%s) is running.", job.Namespace, job.Name)
+			updateStatus(&job.Status, commonv1.JobRunning, commonutil.JobRunningReason, msg)
```

Comment:
Does "!isRunning(job.Status)" indicate the job is pending? Why to update its status to running?

Reply:
I reuse `JobConditionType` from kubeflow and there is no `Pending`.So, I set `RUNNING` to job if the master pod is pending. We can add a `PENDING` status if necessary later.
https://github.com/kubeflow/common/blob/master/pkg/apis/common/v1/types.go#L112-L137